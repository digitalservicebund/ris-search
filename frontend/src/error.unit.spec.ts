import { config, mount } from "@vue/test-utils";
import { describe } from "vitest";
import ErrorPage from "./error.vue";
import { HttpStatusCode } from "axios";

const mountComponent = (statusCode: number) =>
  mount(ErrorPage, {
    props: {
      error: {
        statusCode,
        name: "",
        message: "",
        fatal: false,
        unhandled: false,
        toJSON: () => ({
          message: "",
          statusCode,
        }),
      },
    },
    global: {
      stubs: {
        NuxtLayout: true,
        SimpleSearchInput: true,
      },
    },
  });

describe("error page", () => {
  beforeAll(() => {
    config.global.renderStubDefaultSlot = true;
  });

  afterAll(() => {
    config.global.renderStubDefaultSlot = false;
  });

  it("shows a specific message for Not Found errors (404)", async () => {
    const component = mountComponent(404);
    expect(component.text()).toMatchSnapshot();
  });

  it("shows a generic message for internal server errors (500)", async () => {
    const component = mountComponent(500);
    expect(component.text()).toMatchSnapshot();
  });

  it("shows a more generic message for other errors", async () => {
    const component = mountComponent(HttpStatusCode.ImATeapot);
    expect(component.text()).toMatchSnapshot();
  });
});
