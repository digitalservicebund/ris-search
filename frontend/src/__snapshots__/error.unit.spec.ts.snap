// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`error page > shows a generic message for internal server errors (500) 1`] = `"Es gab leider einen Fehler  Probieren Sie es zu einem späteren Moment noch einmal.Wir haben den Fehler dokumentiert und an unsere Entwickler:innen weitergeleitet."`;

exports[`error page > shows a more generic message for other errors 1`] = `"Es gab leider einen Fehler  Ein unbekannter Fehler ist aufgetreten: 418"`;

exports[`error page > shows a specific message for Not Found errors (404) 1`] = `"Diese Seite existiert nicht  Überprüfen Sie den eingegebenen Link"`;
